libs/executors/include/hpx/executors/parallel_executor.hpp

```diff
@@ -167,6 +168,17 @@ namespace hpx { namespace parallel { namespace execution {
 
         /// \cond NOINTERNAL
 
+        // OneWayExecutor interface
+        template <typename F, typename... Ts>
+        static
+            typename hpx::util::detail::invoke_deferred_result<F, Ts...>::type
+            sync_execute(F&& f, Ts&&... ts)
+        {
+            return hpx::detail::sync_launch_policy_dispatch<
+                launch::sync_policy>::call(launch::sync, std::forward<F>(f),
+                std::forward<Ts>(ts)...);
+        }
+
```

Comment:
Should we fall back on `async_execute(...).get()` for executors that don't have `sync_execute` (as far as I could tell we don't do that now, but I might have mislooked)?

I like this change since explicitly doing `sync_launch_policy_dispatch` avoids all the overhead of launching tasks. I don't mind if we do actually require having `sync_execute` instead of using `async_execute` as a fallback since it avoids nasty performance surprises if one actually expects inline execution with sync.

On the other hand `bulk_sync_execute` does fall back to `bulk_async_execute` and `wait_all`, but that's slightly different since at least parallel execution requires spawning some tasks in any case.

Reply:
> Should we fall back on async_execute(...).get() for executors that don't have sync_execute (as far as I could tell we don't do that now, but I might have mislooked)?

That happens in the customization points. I added this here to ensure that `sync_execute` is really synchronous (and inline) on this executor (which it was not before).

Reply:
> On the other hand bulk_sync_execute does fall back to bulk_async_execute and wait_all, but that's slightly different since at least parallel execution requires spawning some tasks in any case.

I'm not sure if we should implement `bulk_sync_execute` here as (now that we have `sync_execute`). I wouldn't hurt, I suppose.

Reply:
I don't mind either way, and resolving this does not block this PR at all.